crates/services/sync/src/import.rs

```diff
@@ -118,6 +139,7 @@ impl<P, E, C> Import<P, E, C> {
             p2p,
             executor,
             consensus,
+            cache: SharedMutex::new(HashMap::new()),
```

Comment:
So the cache is created here and lives as long as the `Import` task. How long does that task live? I don't see any clearing logic anywhere. Do we risk OOM errors if this runs for too long?

Reply:
There is a clear in the `launch_stream` function that should, from my understanding clear all data but i'm not sure (cf https://github.com/FuelLabs/fuel-core/pull/2361#issuecomment-2417970089)

Reply:
Oh I see now, thanks. Yeah then we pop the ranges once they are consumed, that makes sense. Then we only need to sort out if the ranges are guaranteed to be the same. Otherwise it seems like we'll leak memory.